src/coreclr/jit/objectalloc.cpp

```diff
@@ -597,6 +618,38 @@ void ObjectAllocator::MarkEscapingVarsAndBuildConnGraph()
 
             return Compiler::fgWalkResult::WALK_CONTINUE;
         }
+
+        Compiler::fgWalkResult PostOrderVisit(GenTree** use, GenTree* user)
+        {
+            GenTree* const tree = *use;
+            if (tree->OperIsLocalStore())
+            {
+                GenTreeLclVarCommon* const lclTree = tree->AsLclVarCommon();
+                unsigned const             lclNum  = lclTree->GetLclNum();
+                if (m_allocator->IsTrackedLocal(lclNum) && !m_allocator->CanLclVarEscape(lclNum))
+                {
+                    if ((lclTree->gtFlags & GTF_VAR_CONNECTED) == 0)
+                    {
+                        // This store was not modelled in the connection graph.
+                        //
+                        // If the stored value was was not a stack-viable allocation or null,
+                        // add an edge to unknown source. This will ensure this local does
+                        // not get retyped as TYP_I_IMPL.
+                        //
+                        GenTree* const data = lclTree->Data();
+                        if (!data->IsIntegralConst(0) && (m_allocator->AllocationKind(data) == OAT_NONE))
+                        {
+                            // Add a connection to the unknown source.
+                            //
+                            JITDUMP("V%02u value unknown at [%06u]\n", lclNum, m_compiler->dspTreeID(tree));
+                            m_allocator->AddConnGraphEdge(lclNum, m_allocator->m_unknownSourceLocalNum);
+                        }
+                    }
+                }
+                tree->gtFlags &= ~GTF_VAR_CONNECTED;
+            }
```

Comment:
Just curious -- why is some handling needed here, and some handling in the other place? Why isn't all this handling done from the same walk to avoid carrying this state between them?

Reply:
In preorder, we walk each tree from any lcl var leaf up towards the root, but we don't always reach the root if we can see that the particular leaf escapes or doesn't escape at some intermediate node.

So at store, we don't know if we have built a model for what gets stored until we've finished visiting all the descendant nodes.